### PR TITLE
Remove Default Accessibility Modifiers

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

Removed default accessibility, no `private` modifiers for members, and no `internal` modifiers for global.
Also, moved a few classes out of the nest and into the globe.

### Bugs Fixed ###

- None.

### API Changes ###

None.

### Behavioral Changes ###

None.

### PR Checklist ###

- [X] Has tests (if omitted, state reason in description)
- [X] Has samples (if omitted, state reason in description)
- [X] Rebased on top of master at time of PR
- [X] Changes adhere to coding standard
- [X] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.editorconfig](https://github.com/mattleibow/Caboodle/blob/no-modifiers/.editorconfig) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/FileSystem_Tests.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/DeviceTests/DeviceTests.Shared/FileSystem_Tests.cs) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/Traits.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/DeviceTests/DeviceTests.Shared/Traits.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Connectivity/Connectivity.ios.reachability.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Connectivity/Connectivity.ios.reachability.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/DataTransfer/DataTransfer.ios.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/DataTransfer/DataTransfer.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Geolocation/Geolocation.android.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Geolocation/Geolocation.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Geolocation/Geolocation.ios.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Geolocation/Geolocation.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Permissions/Permissions.android.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Permissions/Permissions.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Permissions/Permissions.shared.enums.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Permissions/Permissions.shared.enums.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Permissions/Permissions.uwp.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Permissions/Permissions.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Types/Shared/PreserveAttribute.shared.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Types/Shared/PreserveAttribute.shared.cs) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C4%5C11%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201804111748275900-175%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=qhR6S4tKqVnwAl0htjkwaf4R4MtZc58WUJyTH35kbJk%3D&st=2018-04-11T17%3A44%3A19Z&se=2018-05-12T17%3A49%3A19Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.editorconfig](https://github.com/mattleibow/Caboodle/blob/no-modifiers/.editorconfig) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/FileSystem_Tests.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/DeviceTests/DeviceTests.Shared/FileSystem_Tests.cs) | :white_check_mark:Succeeded | |
[DeviceTests/DeviceTests.Shared/Traits.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/DeviceTests/DeviceTests.Shared/Traits.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Connectivity/Connectivity.ios.reachability.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Connectivity/Connectivity.ios.reachability.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/DataTransfer/DataTransfer.ios.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/DataTransfer/DataTransfer.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Geolocation/Geolocation.android.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Geolocation/Geolocation.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Geolocation/Geolocation.ios.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Geolocation/Geolocation.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Permissions/Permissions.android.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Permissions/Permissions.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Permissions/Permissions.shared.enums.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Permissions/Permissions.shared.enums.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Permissions/Permissions.uwp.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Permissions/Permissions.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Types/Shared/PreserveAttribute.shared.cs](https://github.com/mattleibow/Caboodle/blob/no-modifiers/Xamarin.Essentials/Types/Shared/PreserveAttribute.shared.cs) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C4%5C11%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201804111750566130-175%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=zYqZmQI8FXTVuATsrRMn7QaRQ24%2BHuC7PfrabCNWbXk%3D&st=2018-04-11T17%3A46%3A47Z&se=2018-05-12T17%3A51%3A47Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: .editorconfig [hunk 1]
```diff
@@ -44,6 +44,7 @@ dotnet_style_qualification_for_event = false:suggestion
 # Use language keywords instead of framework type names for type references
 dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
 dotnet_style_predefined_type_for_member_access = true:suggestion
+dotnet_style_require_accessibility_modifiers = omit_if_default:warning
```
Comment (mattleibow) [inline .editorconfig]: This was just added Jan 31 or VS 15.7, so no effect just yet. We are ahead of the curve here.

**Comment by mattleibow** [comment]
This may help:
https://github.com/xamarin/Essentials/pull/175/files?w=1

**Comment by jamesmontemagno** [comment]
How about these @mattleibow  https://github.com/xamarin/Essentials/blob/master/Xamarin.Essentials/DeviceInfo/DeviceInfo.shared.cs#L56-L78 ?

**Comment by mattleibow** [comment]
@jamesmontemagno moving that question to #167

**Pull request merged**